tests/test_post_gen_project.py

```diff
@@ -4,32 +4,45 @@
 import pytest
 
 sys.path.append(str(Path(__file__).resolve().parents[1]))
-from hooks.post_gen_project import get_repo_dir  # noqa: E402
+from hooks.post_gen_project import (  # noqa: E402
+    copy_journal_template_files,
+    get_repo_dir,
+)
 
 
-def test_with_repo_get_repo_dir(monkeypatch, tmp_path):
-    monkeypatch.setenv("HOME", str(tmp_path))
-    ck_path = tmp_path / ".cookiecutters"
-    ck_path.mkdir()
-    repo_path = ck_path / "scikit-package-manuscript"
-    repo_path.mkdir()
+def test_get_repo_dir(monkeypatch, user_filesystem):
+    home_dir = user_filesystem[0]
+    monkeypatch.setenv("HOME", str(home_dir))
+    repo_path = home_dir / ".cookiecutters" / "scikit-package-manuscript"
 
-    assert get_repo_dir() == repo_path
+    assert repo_path == get_repo_dir()
 
 
-def test_without_repo_get_repo_dir(monkeypatch, tmp_path):
-    monkeypatch.setenv("HOME", str(tmp_path))
-    ck_path = tmp_path / ".cookiecutters"
-    ck_path.mkdir()
-    repo_path = ck_path / "foo"
-    repo_path.mkdir()
-    repo_path = ck_path / "bar"
-    repo_path.mkdir()
-
+def test_without_repo_get_repo_dir(monkeypatch, user_filesystem_without_repo):
+    home_dir = user_filesystem_without_repo[0]
+    monkeypatch.setenv("HOME", str(home_dir))
     with pytest.raises(FileNotFoundError) as exc_info:
         get_repo_dir()
-        print(str(exc_info))
-        assert (
-            str(exc_info) == "couldn't find scikit-package-manuscript,"
-            " but did find foo, bar"
-        )
+    assert "Couldn't find scikit-package-manuscript," in str(exc_info.value)
+
+
+def test_copy_journal_template_files(monkeypatch, user_filesystem):
+    home_dir = user_filesystem[0]
+    monkeypatch.setenv("HOME", str(home_dir))
```

Comment:
In general I would like us to try and use, as a group standard, `mock` rather than monkeypatch.  There is a discussion here: https://github.com/pytest-dev/pytest/issues/4576.    It is not the end of the world, but again, if we all adopt the same approaches and patterns we will end up having more readable and maintainable code...the very essence of scikit-package.